src/bootstrap/compile.rs

```diff
@@ -1161,8 +1161,14 @@ impl Step for Assemble {
             let dst_exe = exe("rust-llvm-dwp", target_compiler.host);
             let llvm_config_bin = builder.ensure(native::Llvm { target: target_compiler.host });
             if !builder.config.dry_run {
-                let llvm_bin_dir = output(Command::new(llvm_config_bin).arg("--bindir"));
-                let llvm_bin_dir = Path::new(llvm_bin_dir.trim());
+                let llvm_bin_dir = if target_compiler.host == build_compiler.host {
+                    let llvm_bin_dir = output(Command::new(llvm_config_bin).arg("--bindir"));
+                    PathBuf::from(llvm_bin_dir.trim())
+                } else {
+                    // if cross-compiling don't copy from the llvm directory for the target instead
+                    PathBuf::from(format!("build/{}/llvm/build/bin", target_compiler.host))
```

Comment:
The build directory might be not here - e.g., ../build or something might be necessary, I think. This also seems fragile if LLVM was supplied externally (whether from CI or via configured llvm-config) - have we tested those cases, or have some reason to expect confidence?

At minimum, IIRC builder.out should work instead of the raw build prefix here, but there maybe a llvm specific one that I can't recall off hand as well.

Reply:
Thank you @Mark-Simulacrum that was actually exactly the implementation help I was looking for in the original issue ❤️